### PR TITLE
feat: distributed authority sample/expanded controls [MTT-9277]

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
This PR adds the skeleton mobile controls UI to the project. Mobile controls can be debugged inside the editor by toggling the following field inside the Bootstrap scene:
Screenshot 2024-11-15 at 16 48 12

A result of this bootstrapping mobile work is that there's a wrapper between the UI and the project's inputs. So, at a high level, UI captures the value of a joystick or a button and routes them to the associated input system action. The result of this is some wrapper Input classes were removed or deprecated, as the game should fetch the input data from the new `GameInput` class, instead of referencing actions independently. The reason for this is that `InputSystemManager` now handles what platform the game is on and subsequently what control scheme to be on. Thus, reading from GameInput will track the currently enabled actions.

Note: having a gamepad connected while debugging mobile controls will override both the virtual joystick inputs, so make sure to test gamepad when outside mobile debug mode.

### Issue Number(s)
[MTT-9277](https://jira.unity3d.com/browse/MTT-9277)

### Contribution checklist
 - [ N/A ] Tests have been added for the project and/or any internal package
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        void SetupControlBindings()
+        {
+            m_ControlMap = new Dictionary<string, InputControl>
+            {
+                { nameof(MobileGamepadState.LeftJoystick), MapRuntimeControl(m_MoveAction) },
+                { nameof(MobileGamepadState.RightJoystick), MapRuntimeControl(m_LookAction) },
+                { nameof(MobileGamepadState.ButtonJump), MapRuntimeControl(m_JumpAction) },
+                { nameof(MobileGamepadState.ButtonInteract), MapRuntimeControl(m_InteractAction) },
+                { nameof(MobileGamepadState.ButtonSprint), MapRuntimeControl(m_SprintAction) },
+                // will re-visit to evaluate if this button is necessary
```
Comment (Elfi0Kuhndorf) [inline Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadBehaviour.cs:98]: do you want to revisit it in a later PR?

File: Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadState.cs [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        /// Event fired when a button bound to the InputSystem is changed.
+        /// </summary>
+        public event Action<string, float> ButtonStateChanged;
+        /// <summary>
+        /// Event fired when a joystick position bound to the InputSystem is changed.
+        /// </summary>
+        public event Action<string, Vector2> JoystickStateChanged;
+
+        /// <summary>
+        /// This method let the UI updates when a property bound with <see cref="BindingMode.ToTarget"/> is calling it.
```
Comment (Elfi0Kuhndorf) [inline Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadState.cs:62]: "lets the UI update"

File: Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadState.cs [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        /// </summary>
+        /// <param name="property">The property bound in the UI using <see cref="BindingMode.ToTarget"/></param>
+        void NotifyUI([CallerMemberName] string property = "")
+        {
+            propertyChanged?.Invoke(this, new BindablePropertyChangedEventArgs(property));
+        }
+
+        /// <summary>
+        /// This method lets the Input system updates when a property value is changed.
```
Comment (Elfi0Kuhndorf) [inline Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadState.cs:71]: same here

File: Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadBehaviour.cs (hunk 1 above)
Comment (fernando-cortez) [reply Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadBehaviour.cs:98]: Yeah, we'll revisit if we need this button in the UI icons PR.

**Comment by Elfi0Kuhndorf** [comment]
The controller controls work well!

With the mobile ones I have some issues:
- The jumping is very hard to execute when walking
- The buttons are too small it is hard to execute them/ not comfortable
- The interact button was for me in the right upper corner, which is not a good position to execute it, it should be close to the jump button

**Comment by fernando-cortez** [comment]
> The controller controls work well!
> 
> With the mobile ones I have some issues:
> 
> * The jumping is very hard to execute when walking
> * The buttons are too small it is hard to execute them/ not comfortable
> * The interact button was for me in the right upper corner, which is not a good position to execute it, it should be close to the jump button

What was challenging about the jumping? Is it something a follow-up PR could address?
If every bullet here could be addressed in a follow-up PR with images/positioning, then I think this one can go through as is.

**Comment by Elfi0Kuhndorf** [comment]
> What was challenging about the jumping? Is it something a follow-up PR could address?

It was hard to actually hit the jump button when walking, it did not execute well for me while I pressed the walking button.

**Comment by hammerlerobi** [comment]
Tested on mobile needs some tweaking, but i guess we will have a follow up

**Review by Elfi0Kuhndorf — APPROVED** [review]
The framing of the main menu on mobile is not perfect (a bit zoomed out would be better), but otherwise it is good to go. I did not encounter any errors, worked as expected. As discussed the buttons should be adjusted in new PR.

**Review by hammerlerobi — APPROVED** [review]

**Review by hammerlerobi — APPROVED** [review]

**Pull request merged**